### PR TITLE
rc: update 1.7.4 -> unstable-2021-08-03

### PR DESCRIPTION
**Pull request by RamKromberg**
###### Description of changes

Updated to latest. Upstream switched to github. Added yacc.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 171301` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>rc</li>
  </ul>
</details>

**Comment by totoroot** [comment]
```
error: Function called without required argument "yacc" at /home/mathym/.cache/nixpkgs-review/pr-171301-1/nixpkgs/pkgs/shells/rc/default.nix:1, did you mean "byacc", "sacc" or "ytcc"?
(use '--show-trace' to show detailed location information)

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/171301

1 package failed to build:
rc

error: build log of '/nix/store/h4ykb9c5xl85x6i6fydwp29k5fxbfzyi-rc-unstable-2021-08-03.drv' is not available
error: build log of '/nix/store/5p6jh94c4n8db399fhagsh36g7wqrf69-rc-unstable-2021-08-03' is not available
```

**Comment by RamKromberg** [comment]
hmm... I guess yacc was deprecated in favor of bison or byacc at some point but I don't understand why it only raises flags during submission and not on my box... well, whatever. I'll resubmit after trying both out first.

**Comment by RamKromberg** [comment]
minor formatting tweaks

File: pkgs/shells/rc/default.nix [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
   configureFlags = [
     "--enable-def-interp=${stdenv.shell}" #183
     ] ++ lib.optionals historySupport [ "--with-history" ]
     ++ lib.optionals (readline != null) [ "--with-edit=readline" ];
 
+  #reproducible-build
   prePatch = ''
     substituteInPlace configure.ac \
-      --replace "date -I" "echo 2015-05-13" #reproducible-build
+      --replace "$(git describe || echo '(git description unavailable)')" "${builtins.substring 0 7 src.rev}"
```
Comment (Mindavi) [inline pkgs/shells/rc/default.nix:28]: Git is not available and fetchFromGitHub doesn't download the full repo by default

File: pkgs/shells/rc/default.nix [hunk 2]
```diff
@@ -1,42 +1,42 @@
-{ lib, stdenv, fetchurl, autoreconfHook
-, ncurses #acinclude.m4 wants headers for tgetent().
-, historySupport ? false
-, readline ? null
-}:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, byacc
+, ncurses # acinclude.m4 wants headers for tgetent().
+, historySupport ? false, readline ? null }:
```
Comment (SuperSandro2000) [inline pkgs/shells/rc/default.nix:3]:
```suggestion
{ lib, stdenv, fetchFromGitHub, autoreconfHook, byacc
, ncurses, readline
, historySupport ? false, readlineSupport ? true }:
```

File: pkgs/shells/rc/default.nix [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
+    owner = "rakitzis";
+    repo = "rc";
+    rev = "8ca9ab1305c3e30cd064290081d6e5a1fa841d26";
+    sha256 = "0744ars6y9zzsjr9xazms91qy6bi7msg2gg87526waziahfh4s4z";
   };
 
   nativeBuildInputs = [ autoreconfHook ];
-  buildInputs = [ ncurses ]
+  buildInputs = [ byacc ncurses ]
     ++ lib.optionals (readline != null) [ readline ];
```
Comment (SuperSandro2000) [inline pkgs/shells/rc/default.nix:18]:
```suggestion
    ++ lib.optionals readlineSupport [ readline ];
```

File: pkgs/shells/rc/default.nix [hunk 4]
```diff
@@ -1,42 +1,42 @@
...
+  src = fetchFromGitHub {
+    owner = "rakitzis";
+    repo = "rc";
+    rev = "8ca9ab1305c3e30cd064290081d6e5a1fa841d26";
+    sha256 = "0744ars6y9zzsjr9xazms91qy6bi7msg2gg87526waziahfh4s4z";
   };
 
   nativeBuildInputs = [ autoreconfHook ];
-  buildInputs = [ ncurses ]
+  buildInputs = [ byacc ncurses ]
```
Comment (SuperSandro2000) [inline pkgs/shells/rc/default.nix:17]:
```suggestion

  # acinclude.m4 wants headers for tgetent().
  buildInputs = [ byacc ncurses ]
```

File: pkgs/shells/rc/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/shells/rc/default.nix:28]: If we really need this use ´´fetchFromGitHub.rev``.

Comment (RamKromberg) [reply pkgs/shells/rc/default.nix:28]: > Git is not available and fetchFromGitHub doesn't download the full repo by default
@Mindavi That was my intention. The ``git describe`` call is used exclusively for setting the ``$version`` environment variable. e.g.
1. execute ``rc``.
2. in rc's shell, run ``echo $version``.

``src.rev`` returns the same hash so I just trimmed it a bit to spare us having to do a full git clone.

> If we really need this use ´´fetchFromGitHub.rev``.

@SuperSandro2000 I'm not sure I'm following. Is something wrong with ``src.rev``?

**Comment by RamKromberg** [comment]
adjusted everything asked except https://github.com/NixOS/nixpkgs/pull/171301#discussion_r867554558 since I'm not sure what to do ¯\_(ツ)_/¯

File: pkgs/shells/rc/default.nix [hunk 5]
```diff
@@ -1,42 +1,44 @@
...
+    ++ lib.optionals readlineSupport [ readline ];
 
   configureFlags = [
     "--enable-def-interp=${stdenv.shell}" #183
     ] ++ lib.optionals historySupport [ "--with-history" ]
-    ++ lib.optionals (readline != null) [ "--with-edit=readline" ];
+    ++ lib.optionals readlineSupport [ "--with-edit=readline" ];
 
+  #reproducible-build
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/shells/rc/default.nix:28]:
```suggestion
  postPatch = ''
```

File: pkgs/shells/rc/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/shells/rc/default.nix:30]: Should be fine like this.

**Comment by RamKromberg** [comment]
thanks :)

**Pull request merged**